### PR TITLE
Remove [BindProperty] on class

### PR DESCRIPTION
**Pull request by rynowak**
This isn't a good fit with consistency with controllers. Discussed with
@DamianEdwards and we agreed to remove this for now and bring it back in
the future if there's a real need for it.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs [hunk 1]
```diff
@@ -239,34 +239,19 @@ internal static PageBoundPropertyDescriptor[] CreateBoundProperties(TypeInfo typ
...
-                if (bindingInfo == null && bindPropertyOnType == null)
+                // property. So we won't bind this property.
+                if (bindingInfo == null)
                 {
                     continue;
                 }
 
-                // If this property is declared as part of a pages base class, then it's likely infrastructure,
-                // so skip it.
-                if (property.Property.DeclaringType.GetTypeInfo().IsDefined(typeof(PagesBaseClassAttribute)))
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs]: Can you update the docs on `PagesBaseClassAttribute`:

> Applying this attribute to a type suppresses discovery of handler methods ~~and bound properties for that type~~.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs]: Will do!

Yeah we don't need that anymore. It's a complicated thing to explain and it made me a little uncomfortable in the first place. So I'm glad to be rid of it.

**Comment by rynowak** [comment]
@pranavkm - I merged this last night, and github didn't update because of the rebase

**Comment by pranavkm** [comment]
Doesn't look like your changes made it to dev - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/BindPropertyAttribute.cs#L9.

**Comment by rynowak** [comment]
wierd.

OK, i'll fix it with maximum force.

**Comment by rynowak** [comment]
Merged

**Pull request closed without merging**